discovery/triton/triton.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2016 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package triton
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"time"
+
+	promlog "github.com/prometheus/common/log"
```

Comment:
So far we are using the global logger mostly. Ideally we want to have structured logging with contextualised loggers being passed down. The Kubernetes SD follows that pattern it would be great if any new SDs would as well: https://github.com/prometheus/prometheus/blob/master/discovery/kubernetes/kubernetes.go#L79